libbindgen/src/codegen/mod.rs

```diff
@@ -324,8 +324,11 @@ impl CodeGenerator for Var {
                 .expr();
             let item = match *val {
                 VarType::Int(val) => {
-                    const_item.build(helpers::ast_ty::int_expr(val))
-                        .build(ty)
+                    let kind = ctx.resolve_item(self.ty()).expect_type().kind();
```

Comment:
Actually we can check more easily in `ir/var.rs`, where we already have the canonical type. There fixing #274 would be also trivial. I guess we'll need to add a `VarType::Bool(bool)` variant for this.